### DISCUSSION TITLE
Seq instances running on AKS with shared storage and scaling

### DISCUSSION THREAD

**Discussion by fpotjens** [datalust/seq-tickets#1657] [Help] [answered]
Hi all,

I was able to setup an AKS cluster with a Seq instance running using a managed premium disk. I used the HELM installation with nginx and was also able to connect and ingest to my seq instance using a DNS entry.
I would like to use the Seq instance/setup for multiple services in one (production) environment. When we have massive logging event (i.e. in case of a failing resource in Azure) I think we need to scale the amount of instances on AKS based on the load and/or amount of requests. Therefor I need a shared storage/volume that will be mounted on each Seq pod instance.... 

I tried to set this up, but eventually it didn't work and then I read the line: 
_... Seq needs a managed disk to store data (**not Azure Files**), which corresponds to the managed-premium storage class. This can be configured when deploying the Helm chart..._

Two questions:
1. Is it possible to set it up this way, or is the Seq server that fast that we don't need it and can just run one instance?
2. In case of failure/restoring the pod/installation, how can I easily backup/restore the managed disk/data?

**Comment by nblumhardt** [comment, marked as answer]
Hi Frank, thanks for reaching out!

It's not possible for multiple Seq instances to share the same storage; we're working towards this kind of feature for the future but today Seq generally scales up rather than out.

Using multiple API keys and controlling levels/ingestion filters on a per-API key basis sounds like the best approach to deal with the situation you're referring to, though - in the case of a log flood, the flood can be suppressed by filtering, and in the meantime, Seq is generally happy to ingest high volumes to a single node (the main impact will be on the query side).

Setting up a second node for [disaster recovery](https://docs.datalust.co/docs/setting-up-a-disaster-recovery-dr-instance) is the easiest way to mitigate against a full node failure and redundantly store log data; in other cases, using [Backup and Restore](https://docs.datalust.co/docs/backup-and-restore) will sort out configuration recovery.

Let me know if you need more detail on any of the above.

Best regards,
Nick

**Reply by fpotjens** [reply]
Hi Nick,

Thanks for the clear answer! Some remarks/questions from my side.

> It's not possible for multiple Seq instances to share the same storage; we're working towards this kind of feature for the future but today Seq generally scales up rather than out.

Is there some kind of global planning/roadmap when you guys are aiming to implement AKS scaling (with shared storage) into the Seq solution? AKS does support (shared) volumes based on Azure File Storage that can be mounted during the deployment of a pod on the nodepool. In our company we already use this solution and works perfectly fine when scaling out! 👍 

> Setting up a second node for [disaster recovery](https://docs.datalust.co/docs/setting-up-a-disaster-recovery-dr-instance) is the easiest way to mitigate against a full node failure and redundantly store log data; in other cases, using [Backup and Restore](https://docs.datalust.co/docs/backup-and-restore) will sort out configuration recovery.

Is this also supported within the AKS solution? I mean, I think I can accomplish this by deploying a separate instance of Seq (with different name) and his own managed premium storage attached. I have connected an public IP address to the first (primary) instance (via the AKS load balancer of nginx), but I'm not sure if it's possible to extend the load balancer with the extra pod instance (the DR instance). When looking at the documentation about Setting up a Disaster Recovery (DR) Instance, I only see a situation based on a IaaS solution with VMs. 

Looking forward to your answers and thoughts!

Best regards,
Frank

**Reply by liammclennan** [reply]
Hi Frank,

If/when Seq supports scaling out it will probably involve non-shared storage as a way of scaling IO. Databases generally don't scale out with shared storage because they would immediately experience a bottleneck on storage, especially in the cloud where IO performance is already a challenge. 

The write side of Seq is quite fast and mostly limited by the available IO. If you are worried about write performance make sure you use a high performance disk. 

We don't have a lot of experience with AKS, but I'd be very surprised if it can't host a Seq DR cluster. Please note that DR is only available in Seq 2021.3 or later.

Let us know how you go or if we can help further.

**Reply by fpotjens** [reply]
Hi Liam,

Thanks for that reply! We will play around with setting up a DR instance on AKS. Also good to know that we don't have to expect shared storage on different nodes, that changes the scalability/performance plan we will need to create.

Thanks again!

Regards,
Frank
